core/modules/replica/ClusterHealthJob.h

```diff
@@ -0,0 +1,257 @@
+/*
+ * LSST Data Management System
+ * Copyright 2018 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+#ifndef LSST_QSERV_REPLICA_CLUSTERHEALTHJOB_H
+#define LSST_QSERV_REPLICA_CLUSTERHEALTHJOB_H
+
+/// ClusterHealthJob.h declares:
+///
+/// class  ClusterHealthJob
+///
+/// (see individual class documentation for more information)
+
+// System headers
+#include <atomic>
+#include <functional>
+#include <map>
+#include <string>
+#include <vector>
+
+// Qserv headers
+#include "replica/Job.h"
+#include "replica/ServiceManagementRequest.h"
+#include "replica/TestEchoQservMgtRequest.h"
+
+// This header declarations
+
+namespace lsst {
+namespace qserv {
+namespace replica {
+
+/**
+ * This data structure represents a summary of services within a cluster
+ * s reported by the job upon its completion.
+ */
+class ClusterHealth {
+
+public:
+
+    /**
+     * Normal (and the only) constructor requires to know names
+     * of all workers contributing into the report.
+     *
+     * @param workers - names of the workers
+     */
+    ClusterHealth(std::vector<std::string> const& workers);
+
+    // Default construction and copy semantics are prohibited
+
+    ClusterHealth() = delete;
+    ClusterHealth(ClusterHealth const&) = delete;
+    ClusterHealth& operator=(ClusterHealth const&) = delete;
+
+    ~ClusterHealth() = default;
+
+    /// @return 'true' if all services are up
+    bool good() const { return _good; }
+
+    /// @return status of the Replication worker services ('true' if responded)
+    std::map<std::string, bool> const& replication() const { return _replication; }
```

Comment:
I guess the members of _replication are inserted in the ctor, so there's no possibility of collision there. I can't see where _qserv is used (except in `qserv()`) so I don't know about that one.

Reply:
Yes, thread safety is not a concern here because objects of this type are constructed and returned by the synchronized method:
`/**`
`     * @return summary report`
`    *`
`     * @throw std::logic_error if the method is called before the job finishes`
 `    */`
`    ClusterHealth const& clusterHealth() const;`